sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -414,91 +396,5 @@ public override void Dispose()
 
             public override string ToString() => _responseMessage.ToString();
         }
-
-        private class ContentStream : ReadOnlyStream
```

Comment:
What was the purpose of this type in the first place?

Reply:
Make content stream access lazy, idea was that we can postpone calling ReadAsStreamAsync until a read on the stream is called. Unfortunately in most cases a sync stream access would be performed first.